### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by malikhaiderkhan**
In this project I:

- Create a class Classroom.
- Create the has-many/belongs-to relationship between Classroom and Student.
- Create a class Book.
- Create a class Rental.
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rentals << rental
+  end
```
Comment (zeff96) [inline book.rb]: - [x] This approach lacks clear associations between Person, Book, and Rental, making it challenging to maintain a many-to-many relationship. I recommend considering the below suggestion, which encapsulates rental creation within the Book class and properly sets associations, improving code clarity and maintainability. 🔄
```suggestion
 def add_rental(person, date)
    `# *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.`
    Rental.new(date, self, person)
  end
```

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (zeff96) [inline book.rb]: - [ ]Getters and Setters for @rentals should be present
```suggestion
  `attr_accessor :title, :author, :rentals`
```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (zeff96) [inline rental.rb]: - [ ] Getters and Setters for @book and @person instance variables should be present
```suggestion
  attr_accessor :date, :book, :person
```

File: person.rb [hunk 4]
```diff
@@ -3,7 +3,7 @@
 require './trimmer_decorator'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rental
```
Comment (zeff96) [inline person.rb]: - [ ] Getters and Setters for @rentals should be present
```suggestion
  `attr_accessor :name, :age, :rentals`
```

File: person.rb [hunk 5]
```diff
@@ -22,6 +23,10 @@ def correct_name
     @name
   end
 
+  def add_rental
```
Comment (zeff96) [inline person.rb]: - [ ] This approach lacks clear associations between Person, Book, and Rental, making it challenging to maintain a many-to-many relationship. I recommend considering the below suggestion, which encapsulates rental creation within the Person class and properly sets associations, improving code clarity and maintainability. 🔄
```suggestion
  ```ruby
def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```
```

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️ ♻️ ♻️ 
Hi @malikhaiderkhan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

close

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Correct usage of GitFlow


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by malikhaiderkhan** [comment]
Fix all the suggestions as suggested by the reviewer.

**Review by KDlamini — APPROVED** [review]
### Hi @malikhaiderkhan 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for setting up associations. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Professional pull request
✔️ Well-documented and professional README
✔️ There are `has-many` / `belongs-to` relationships
✔️ Associations are well implemented where applicable

# Status: Approved ✔️ 

oscars-standing-ovation

## Optional Suggestions

- In the future, remember to always follow the proper guidelines for writing profession commit messages. In your case, it would be best to begin every message with a capital letter as you would with any sentence.

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**